app/controllers/api/v1/deploys_controller.rb

```diff
@@ -0,0 +1,48 @@
+class Api::V1::DeploysController < Api::V1::ApplicationController
+  include CurrentProject
+
+  before_filter :authorize_project_deployer!
```

Comment:
something simple like a module should work for starters ... the only duplication would be the before_filter calls ...
the json rendering could be done by handlin all response types in that module too like request.json -> unprocessable_entity etc ... or raising a error and then using rescue_from and show the error message as json in the api ... 

for the authorize_project_deployer a `authorize_deploy_modification :create` helper could do the trick until we come up with something nicer